src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -97,15 +107,30 @@ protected override void OnNextCore(Block block)
 
             this.logger.LogTrace("Block header '{0}' added to the announce queue.", chainedBlock);
             this.blocksToAnnounce.Enqueue(chainedBlock);
+            this.blockEnqueued.Set();
+
+            this.logger.LogTrace("(-)");
+        }
+
+        public void Initialize()
+        {
+            this.logger.LogTrace("()");
+
+            this.relayWorkerTask = Task.Run(async delegate { await this.RelayWorkerAsync(); });
```

Comment:
We should avoid Task.Run in backround work thats why we have the async loop.
I know in the peer code we dont use async loop but that is an exception.

Reply:
i can't see how to use async loop easily here, async loop can't be called currently on demand

Reply:
Your are effectively starting a background loop, so you use async loop with the run once property.

Reply:
That does not make much sense. Instead of simple Task.Run (or not even that as I suggest) you want to construct AsyncLoop that won't even loop? Why would you do that?

Reply:
Constructing an asyncloop is not so hard and that's the pattern we use, it makes it easier to dispose of the loop and puts it under our control exception handling, life time etc...

When you just do a Task,Run especially on a standalone loop it become to top long running task, you then need to manage exceptions and manage the loop.

Also see my comment on the issue we could just reduce the loop time and still achieve significant improvement.

Reply:
Fixed.

Reply:
>Also see my comment on the issue we could just reduce the loop time and still achieve significant improvement.

This will force us to send smaller batches which creates way more overhead which is unnecessary.

Reply:
well I suggest not to construct async loop or Task.Run
but for single run I would go with Task.Run

Reply:
> This will force us to send smaller batches which creates way more overhead which is unnecessary.

i don't agree, the alg is designed to send batch if there is a tip (which is a must imho) or to wait 5 seconds and send whatever we have if nothing new comes

Reply:
Ok, I'll not push new commits until we all come to consensus on what to do =)

Though I also think that creating an AsyncLoop instance for a loopless task lacks logic. But it looks like we do it all around the codebase so if we agree to remove this one we also should do it for other usages of the AsyncLoop that create RunOnce 'loops'.